openapi_tester/validators.py

```diff
@@ -74,24 +73,64 @@ def wrapped(value: Any):
 def validate_type(schema_section: Dict[str, Any], data: Any) -> Optional[str]:
     schema_type: str = schema_section.get("type", "object")
     if not VALIDATOR_MAP[schema_type](data):
+        an_articles = ["integer", "object", "array"]
         return VALIDATE_TYPE_ERROR.format(
-            expected=OPENAPI_PYTHON_MAPPING[schema_type],
-            received=type(data).__name__,
+            article="a" if schema_type not in an_articles else "an",
+            type=schema_type,
+            received=f'"{data}"' if isinstance(data, str) else data,
         )
     return None
 
 
 def validate_format(schema_section: Dict[str, Any], data: Any) -> Optional[str]:
     schema_format: str = schema_section.get("format", "")
     if schema_format in VALIDATOR_MAP and not VALIDATOR_MAP[schema_format](data):
-        return VALIDATE_FORMAT_ERROR.format(expected=schema_section["format"], received=str(data))
+        example_values = {
```

Comment:
I'd completely remove the example.

The only think that might be needed here is a map, but really I think we can also do without the article :),

expected: "email" formatted value
received: {value}

This is simple, low maintainance and we lose no cohesion in doing this.

Reply:
Yeah lets do this 👍